File: programmers/greedy/lev2_구명보트.py
# 무게한도가 존재하는 보트로 사람들을 옮길 수 있는 최소한의 횟수를 구하는 문제.

def solution(people, limit):
    
    end = 1
    start = 0
    boat = 0
    speople = sorted(people)
    while True:
        if end + start > len(speople):
            break
        sum = 0
        if end+start != len(speople):
            sum = speople[-end] + speople[start]
        else:
            boat = boat + 1
            break
        if sum <= limit :
            boat = boat + 1
            end = end + 1
            start = start + 1
        else:
            boat = boat + 1
            end = end + 1
        
            
    answer = boat
    return answer

File: python/programmers/greedy/lev2_구명보트.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <iostream>

using namespace std;

int solution(vector<int> people, int limit) {
    int answer = 0;
    int boat = 0;
    int size = people.size();
    int start = 0;
    int end = 0;
    
    sort(people.begin(), people.end());
    
    while(start + end < size){
        boat = limit;
        boat -= people[size-end-1];
        end++;
        if(people[start] <= boat) {
            start++;
        }
        answer++;
    }
    
    return answer;
}
